src/Host/Broker/Impl/Sessions/SessionManager.cs

```diff
@@ -59,6 +60,41 @@ ILogger<Process> hostOutputLogger
             }
         }
 
+        public void CloseAndBlockSessionsCreationForUser(IIdentity user) {
+            lock (_sessions) {
```

Comment:
I'd rather use a single lock for both collections, to reduce the potential for wrong order deadlock bugs. It would lock for longer than it has to, but session creation is not an operation that happens frequently, so this shouldn't be a perf concern.